### PR TITLE
just 0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by Ezku**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Hello,

I had need of `just` and found it it was not on Homebrew yet. Hope this PR looks good to you. The process seemed straightforward enough, thanks for the good work on documentation.

Cheers,
Eevert

### CODE REVIEW

File: Formula/just.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Just < Formula
+  desc "🤖 Just a command runner"
+  homepage "https://github.com/casey/just"
+  url "https://github.com/casey/just.git",
```
Comment (DomT4) [inline Formula/just.rb]: Does this not build if you use the archive tarball?

File: Formula/just.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Just < Formula
+  desc "🤖 Just a command runner"
```
Comment (DomT4) [inline Formula/just.rb]: Heh, probably not the place for emojis, however fond of them I am personally.

File: Formula/just.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--release"
+    bin.install "target/release/just"
+  end
+
+  test do
+    system "#{bin}/just", "--version"
```
Comment (DomT4) [inline Formula/just.rb]: Any chance of a better test here? Looking at the upstream `README` it looks like something more functionality-based could be done?

Comment (Ezku) [reply Formula/just.rb]: Let's see, I got a hint from the author on this: https://github.com/casey/just/issues/153#issuecomment-335925847

File: Formula/just.rb (hunk 1 above)
Comment (Ezku) [reply Formula/just.rb]: That's what I tried at first, but with no success.

Comment (Ezku) [reply Formula/just.rb]: My mistake was that I had the wrong tarball cached, so the build step would always fail.

```
==> Downloading https://github.com/casey/just/archive/v0.3.1.tar.gz
Already downloaded: /Users/ezku/Library/Caches/Homebrew/just-0.3.1.tar.gz
==> cargo build --release
Last 15 lines from /Users/ezku/Library/Logs/Homebrew/just/01.cargo:
2017-10-13 14:56:03 +0300

cargo
build
--release

error: could not find `Cargo.toml` in `/private/tmp/just-20171013-31183-1fb8xx3` or any parent directory
```

The dist release of just, with only the prebuilt binary, had somehow ended up in the cache under the same name as the github source archive tarball. Deleting the file worked.

**Comment by Ezku** [comment]
@DomT4, thanks for the review, I've tried to address your concerns. How would I request another round? :)

**Comment by valscion** [comment]
Seems like the description needs tuning to satisfy the style error:

screen shot 2017-10-13 at 15 25 39

**Comment by Ezku** [comment]
My bad, I only checked `brew audit --strict`

**Review by valscion — APPROVED** [review]

**Comment by DomT4** [comment]
Could perhaps squash the commits together for neatness, but in terms of pure syntax it looks okay to me. Will hand it over to ILZ to review.

**Comment by casey** [comment]
Just pinging this thread: Is this ready to merge? It looks like @valscion has approved it.

**Comment by DomT4** [comment]
Needs a review from @ilovezfs. He's the big cheese.

**Comment by dunn** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by dunn** [comment]
Merged as 739888e36e25bb62e4a816effdde73230f167a06, thank you for contributing to Homebrew!

**Pull request closed without merging**